src/applications/disability-benefits/all-claims/components/OfficialReport.jsx

```diff
@@ -0,0 +1,304 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { SchemaForm } from 'platform/forms-system/exportsFile';
+import FormNavButtons from '~/platform/forms-system/src/js/components/FormNavButtons';
+import { useEditOrAddForm } from 'platform/forms-system/src/js/patterns/array-builder';
+import ArrayBuilderCancelButton from 'platform/forms-system/src/js/patterns/array-builder/ArrayBuilderCancelButton';
+import { getArrayUrlSearchParams } from 'platform/forms-system/src/js/patterns/array-builder/helpers';
+import {
+  VaModal,
+  VaButton,
+  VaAlert,
+} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+import {
+  officialReport,
+  officialReportMst,
+} from '../pages/form0781/officialReport';
+import { removePoliceReportModalContent } from '../content/officialReport';
+import { isRelatedToMST } from '../utils/form0781';
+import { POLICE_REPORT_LOCATION_FIELDS } from '../constants';
+
+export function shouldShowPoliceDataModal(event) {
+  return (
+    event &&
+    !event?.otherReports?.police &&
+    POLICE_REPORT_LOCATION_FIELDS.some(
+      field => typeof event[field] === 'string' && event[field].trim() !== '',
+    )
+  );
+}
+
+const OfficialReport = props => {
+  const formDomRef = useRef(null);
+  const modalRef = useRef(null);
+  const alertRef = useRef(null);
+  const submitButtonRef = useRef(null);
+
+  const searchParams = getArrayUrlSearchParams();
+  const isEdit = !!searchParams.get('edit');
```

Comment:
Non blocking question: Is this a workaround related to the Array Builder? This is getting query strings directly from the url yeah?

Reply:
[This](https://github.com/department-of-veterans-affairs/vets-website/blob/38c2e960a3651dde4019a7a7e363cb1a381f37d1/src/platform/forms-system/src/js/patterns/array-builder/ArrayBuilderItemPage.jsx#L33) is how the `ArrayBuilderItemPage` is constructed, and from which I was advised to base the reimagining of the Official Report page. It is, admittedly, a bit of a hack to determine which state the item page is in, to facilitate conditional logic of each page state.